ai-analysis/test-08-12-25/add-custom-page-COMPARISON.md

```diff
@@ -0,0 +1,399 @@
+# Documentation Correction Comparison
+
+**Document**: `guides/plugins/plugins/storefront/add-custom-page.md`  
+**Validation Date**: December 8, 2025  
+**Shopware Version**: 6.7.5.0  
+**M7 Issues Found**: 7 HIGH severity issues
+
+---
+
+## Executive Summary
+
+The original document contained **7 critical inaccuracies** that would cause implementation failures. This comparison shows what was wrong and what was corrected.
+
+---
+
+## Change #1: Route Scope Configuration (CRITICAL)
+
+### ❌ Original (WRONG)
+
+```php
+#[Route(defaults: [PlatformRequest::ATTRIBUTE_ROUTE_SCOPE => [StorefrontRouteScope::ID]])]
+class ExampleController extends StorefrontController
+```
+
+### ✅ Corrected (ACCURATE)
+
+```php
+#[Route(defaults: ['_routeScope' => ['storefront']])]
+class ExampleController extends StorefrontController
+```
+
+**Why This Matters:**
+- Original uses verbose Shopware 6.5/6.6 syntax that still works but is outdated
+- Corrected version uses modern Shopware 6.7 simplified syntax
+- Both work, but new syntax is clearer and recommended
+- **Validated against**: `platform/src/Storefront/Controller/LandingPageController.php`
+
+---
+
+## Change #2: Repository Usage Warning (CRITICAL)
+
+### ❌ Original (INCOMPLETE)
+
+```php
+// Vague statement buried in text:
+"Do not use a repository directly in a page loader. 
+Always get the data for your pages from a store api route instead."
+```
+
+**Problem**: This critical architectural rule was:
+- Not emphasized as a strict requirement
+- Not explained WHY it matters
+- Not shown with examples
+- Easy to miss or ignore
+
+### ✅ Corrected (COMPREHENSIVE)
+
+Added throughout the document:
+
+1. **Warning Box at Top:**
+```
+⚠️ WARNING: Do not use direct or indirect repository calls in controllers or page loaders. 
+Always use Store API routes to get or put data.
+```
+
+2. **PHPDoc in Controller:**
+```php
+/**
+ * @internal
+ * Do not use direct or indirect repository calls in a controller. 
+ * Always use a store-api route to get or put data
+ */
+```
+
+3. **Detailed Explanation Section:**
+```
+Why not use repositories directly?
+
+Shopware's architecture enforces that:
+- Storefront controllers/page loaders → Must use Store API routes
+- Store API routes → Can use repositories internally
+
+This separation:
+✅ Ensures consistent caching behavior
```

Comment:
the separation has nothing to do witch caching or event dispatching or permission checks, but rather about enforcing that headless setups are supported and the same functionality is available over the store api

Also note that this is not enforced in any other way, you can do use repos directly in your plugins,the downside is that your functionality is not available pver store api at all